### PR TITLE
Fix:  Copying the messages and marking a message in between as unread generates 'new' in the text

### PR DESCRIPTION
**Pull request by Victor-Nyagudi**
### Details

Added the `styles.userSelectNone` style to `UnreadActionIndicator` so users are visually aware it's not copied to the clipboard and then removed all elements with this style applied from the copied contents inside `SelectionScraper`. 

This way, anything that's given a style of `user-select: none` is omitted from the clipboard whenever the user copies something.

### Fixed Issues

$ https://github.com/Expensify/App/issues/17838
$ https://github.com/Expensify/App/issues/17838#issuecomment-1523540865

### Tests

1. Log in with an account that has at least one chat created on Mac Chrome/Mac Safari/Mac Desktop
2. Click on any chat
3. Send multiple messages e.g. 1, 2, 3, 4, 5, 6
4. Select a message in between e.g. 4 and mark it as unread (make sure a green line appears after marking as unread, otherwise, try refreshing the page and mark as unread again)
5. Copy all the messages i.e. left-click and drag over all the messages then copy (<kbd>Cmd</kbd>/<kbd>Ctrl</kbd> + <kbd>C</kbd>)
6. Verify the word `New` that's to the right of the green line is not highlighted
7. Paste the selection in the composer i.e. <kbd>Cmd</kbd>/<kbd>Ctrl</kbd> + <kbd>V</kbd>
8. Verify the word `New` is not part of pasted content
9. Click on a separate chat and then come back to the one with unread message to update its status to `read`
10. Copy and paste all the messages into the composer like before
11. Verify that pasted content is the same as what's copied i.e. works as usual

- [x] Verify that no errors appear in the JS console

### Offline tests

Same as above but go offline after logging in i.e. step 1 (or force offline in staging). 

### QA Steps

Same as Tests.

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://user-images.githubusercontent.com/79470910/235470774-5a671f3e-1e88-4516-a598-69a9ab23d4ff.mov

https://user-images.githubusercontent.com/79470910/235470793-b179c611-2bf9-48fb-b4cc-3d6a0f3e5378.mov

</details>

<details>
<summary>Mobile Web - Chrome</summary>

expensify-mobile-chrome-web-screenshot

</details>

<details>
<summary>Mobile Web - Safari</summary>

expensify-mobile-safari-web-screenshot

</details>

<details>
<summary>Desktop</summary>

https://user-images.githubusercontent.com/79470910/235470710-3172aff0-57a0-4a5d-b038-9ce6d58cdd96.mov

</details>

<details>
<summary>iOS</summary>

expensify-ios-native-screenshot

</details>

<details>
<summary>Android</summary>

expensify-android-native-screenshot

</details>

### CODE REVIEW

**Comment by Victor-Nyagudi** [comment]
I have read the CLA Document and I hereby sign the CLA

**Review by fedirjh — DISMISSED** [review]
LGTM 🚀 

cc @Beamanator All yours

**Review by Beamanator — DISMISSED** [review]
LGTM 💪

File: src/components/UnreadActionIndicator.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ import Text from './Text';
 import withLocalize, {withLocalizePropTypes} from './withLocalize';
 
 const UnreadActionIndicator = props => (
-    <View accessibilityLabel={props.translate('accessibilityHints.newMessageLineIndicator')} data-action-id={props.reportActionID} style={styles.unreadIndicatorContainer}>
+    <View accessibilityLabel={props.translate('accessibilityHints.newMessageLineIndicator')} data-action-id={props.reportActionID} style={[styles.unreadIndicatorContainer, styles.userSelectNone]}>
```
Comment (Beamanator) [inline src/components/UnreadActionIndicator.js:8]: Shoot, looks like this is slightly too long 🤷 

```suggestion
    <View
        accessibilityLabel={props.translate('accessibilityHints.newMessageLineIndicator')}
        data-action-id={props.reportActionID}
        style={[styles.unreadIndicatorContainer, styles.userSelectNone]}
    >
```

**Review by Beamanator — APPROVED** [review]

**Comment by AndrewGable** [comment]
Hey all - This caused a deploy blocker here: https://github.com/Expensify/App/issues/18376

I reverted it here: https://github.com/Expensify/App/pull/18380

**Comment by Victor-Nyagudi** [comment]
@AndrewGable I'm currently investigating the unintended behavior.

Thanks.

**Comment by Victor-Nyagudi** [comment]
I was just about to post my findings as you posted yours, @fedirjh. 

### Problem

I also discovered there are a couple of other components that have the `user-select: none` style applied. Some of these are an ancestor with a considerable number of children, grandchildren, etc., so when we remove any `div` with this style applied, the component and all its descendants are removed. 

One of the bigger components being `OfflineWithFeedback`.

https://github.com/Expensify/App/blob/b6eb637298f77156c9dc9c482e30d77b105d2a26/src/components/OfflineWithFeedback.js#L76

https://github.com/Expensify/App/blob/b6eb637298f77156c9dc9c482e30d77b105d2a26/src/components/Reactions/MiniQuickEmojiReactions.js#L75

https://github.com/Expensify/App/blob/b6eb637298f77156c9dc9c482e30d77b105d2a26/src/components/OptionRow.js#L160


### Solution

I explored two possible solutions with the first being the one I went with, and the second being one that could require a bit more work but is more future-proof, however, I'd require some input on it first. 

- Option 1

From my search using VS Code's search tool, I discovered the only component using `accessibilityHints.newMessageLineIndicator` is [`UnreadActionIndicator`](https://github.com/Expensify/App/blob/b6eb637298f77156c9dc9c482e30d77b105d2a26/src/components/UnreadActionIndicator.js#L8), and the only component using `UnreadActionIndicator` is [`ReportActionItem`](https://github.com/Expensify/App/blob/b6eb637298f77156c9dc9c482e30d77b105d2a26/src/pages/home/report/ReportActionItem.js#L324).

Given only one message can be marked unread at a time and the single use-case scenario mentioned above, I opted to just grab the `div` with this accessibility label (localized, of course) and removed it from the copied items inside `SelectionScraper`.  

```javascript
// Get the div housing the UnreadActionIndicator and remove it from copied content
div.querySelector(`[aria-label="${Localize.translateLocal('accessibilityHints.newMessageLineIndicator')}"]`)
    .remove();
```

Should things change in future and multiple messages can be marked as unread, we can update this to `div.querySelectorAll()` to cover all those other unread message indicators. 

Here are a couple of videos with my changes applied. 

<details>

<summary>Mac Chrome English</summary>


https://user-images.githubusercontent.com/79470910/236211579-9efc6561-b685-4384-9305-3cf78370ade1.mov



</details>

<details>

<summary>Mac Chrome Spanish</summary>


https://user-images.githubusercontent.com/79470910/236211605-ec695971-a09a-4c9e-885b-625a35a31ecc.mov



</details>

<details>

<summary>Mac Safari English</summary>


https://user-images.githubusercontent.com/79470910/236211634-b280893c-293c-4bb1-adcd-6b92aab5a7cf.mov



</details>

<details>

<summary>Mac Safari Spanish</summary>


https://user-images.githubusercontent.com/79470910/236211662-37c9fd70-7270-4b03-a0b4-4e982ce31bef.mov



</details>

<details>

<summary>Mac Desktop English</summary>


https://user-images.githubusercontent.com/79470910/236211691-ed8a577e-bf25-457d-b240-d5cad42a0fa8.mov



</details>

<details>

<summary>Mac Desktop Spanish</summary>


https://user-images.githubusercontent.com/79470910/236211708-329a6fc2-9af4-4d24-96e9-2fceda6f2042.mov


</details>

This is a very specific fix, though, and only covers `UnreadMessageIndicator`, but given its very limited use, there's a lower risk of regressions occurring.

As you can see, the emojis are now being copied again. In the previous solution in this PR that caused a regression, the emojis weren't being copied, so it looks like we may have been onto something in possibly fixing other [issues involving copying emojis](https://github.com/Expensify/App/issues/15810) or [inadvertently copying text](https://github.com/Expensify/App/issues/15194).

On the other hand, I don't think it would be a bad idea to focus on one issue at a time, and then propose solutions to the other copying issues once this one is fixed with no regressions. 

- Option 2 (Requires input)

I dabbled with the idea of components that should not be copied receiving a `shouldElementBeCopiedToClipboard` prop that would determine whether or not the component gets an attribute like `data-shouldElementBeCopiedToClipboard="false"`.

We could then go through all the elements with this attribute set to `true` like before and remove them, but I felt like this would require a bigger change in terms of adding props to the component in the pages it's been used. 

```javascript
div.querySelectorAll(`div[data-shouldElementBeCopiedToClipboard="${shouldElementBeCopiedToClipboard}"]`)
    .forEach(item => item.remove());
```

I like [@fedirjh proposal](https://github.com/Expensify/App/pull/18228#issuecomment-1534672831) though. We could also go with that. [Support for `:has()` has improved a lot](https://caniuse.com/?search=has), but it still needs to be enabled on Firefox i.e. not supported by default. 

cc: @Beamanator 

P.S. Notice how in the videos I've shared in this comment how copying the messages from the left side to right doesn't copy the emojis (text cursor is visible) but copying from the right side to left (default cursor) does. 

Just some food for thought going forward.

**Comment by bernhardoj** [comment]
@fedirjh @Victor-Nyagudi 
Just sharing my opinion here.

I previously mentioned that this bug is a browser bug here https://github.com/Expensify/App/issues/15194#issuecomment-1470361112, but looks like everyone disagree with it (the bug isn't reproducible in Firefox). You can found the Chromium bug report from my comment. At first, I thought it's already fixed on the latest version of Chrome, but turns out it's still reproducible and also the commits getting reverted because of a [regression](https://bugs.chromium.org/p/chromium/issues/detail?id=850685#c23).

I think the [commit](https://chromium.googlesource.com/chromium/src/+/97ba01b40f88c678c8bedfc795e1cd14aee97d2b) could be useful especially the test case. If I understand @fedirjh solution correctly, it will select the non selectable text for this case:
```
<div style="user-select: none">
    SKIP
    <div style="user-select: text">
        KEEP
    </div>
</div>
```
Result: SKIP KEEP

However, based on the chromium test case, only KEEP text should be copied. Here is one of the test case example on nested user-select
image

I think it would require a lot of changes to cover all cases without any regression. Even the chromium commits getting reverted 🤣. So, if I can suggest, I would go for the solution to only exclude the text on a specific component in the meantime, in this case the new message indicator, while waiting the full fix from Chromium, which I guess will be taking very long, not to mention Safari.

**Comment by Victor-Nyagudi** [comment]
I'm a little confused here. 

[This PR was reverted](https://github.com/Expensify/App/pull/18380) by @AndrewGable but has been deployed to production. 

Is it the reverted PR that's been deployed to production and we're just being notified, or did something happen in between?

**Comment by Beamanator** [comment]
@Victor-Nyagudi yeah i can see that it's confusing, I looked at the changes made by the revert PR, and they're what's currently in the `production` branch

**Comment by Beamanator** [comment]
> Thank you for the thorough investigation, @bernhardoj. Your explanation is very helpful. I suggest that we go with @Victor-Nyagudi's option 1, which will only exclude the text on a specific component, `UnreadActionIndicator`. We attempted to make a global fix for this issue, but it seems that it is not currently feasible. @Beamanator, what are your thoughts on this? What should be our next steps? Should we go with [this option](https://github.com/Expensify/App/pull/18228#issuecomment-1534797205), or should we revisit the main issue and search for another global fix that addresses all cases?

@fedirjh I agree with this conclusion 👍 

Since @Victor-Nyagudi is the author of this PR (which was reverted), @Victor-Nyagudi can you please make the fix PR & link it to the original issue & mention this one as well?

**Comment by Victor-Nyagudi** [comment]
@Beamanator Got it. 

I'll get to work on the PR. 

Glad that's cleared up.

**Comment by Beamanator** [comment]
Thanks @Victor-Nyagudi 👍

**Comment by Victor-Nyagudi** [comment]
[PR](https://github.com/Expensify/App/pull/18684) is ready for review. 

Let me know if there's anything else you need me to do. 

cc: @fedirjh @Beamanator

**Pull request merged**